Add common store for subject api

tranVyTDT opened the pull request.

tranVyTDT committed:
[VyTT][CreatePost] modify create post view

tranVyTDT committed:
[VyTT][CreatePost] rename props

tranVyTDT committed:
[VyTT][CreatePost] rename constant, and use InputNumber

tranVyTDT committed:
[VyTT][createPost] add Toast

tranVyTDT committed:
[VyTT][createPost] remove Toast

tranVyTDT committed:
Merge branch 'main' into add-common-store-for-subject-api

tranVyTDT committed:
[VyTT][createPost] jump tp how page after submit post

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -1,40 +1,46 @@
 
     const store = commonStore();
+    const isFormSubmitted = ref(false);
+    const isSubjectSelected = ref(true);
```
```suggestion
    const subjectSelected =  computed(() => {
      return Object.keys(postData.value.subjects)
        .filter((key, index) => postData.value.subjects[index].checked);
    })
    const isSubjectSelected = computed(() => {
      return subjectSelected.length > 0;
    })
```

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -44,33 +50,60 @@
+                return false
+        }
+        return true
+    }
```
Fix function validateAgeRange to this
```suggestion
    function validateAgeRange() {
        if (postData.value.requestTeacherAgeFrom || postData.value.requestTeacherAgeTo ) {
            validationMessage.value = 'Tuổi bên trái và tuổi bên phải phải được chọn cả 2';
                return false
        }
        if (postData.value.requestTeacherAgeFrom >= postData.value.requestTeacherAgeTo) {
            validationMessage.value = 'Tuổi bên trái phải nhỏ hơn Tuổi bên phải';
            return false
        } 
        validationMessage.value = '';
        return true
    }
```

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -44,33 +50,60 @@
+    }
+
+    function isShiftValid(shift:ShiftInterface) {
+        return shift.weekday && shift.startTime && shift.endTime ? true : false
```
```suggestion
        return shift.weekday && shift.startTime && shift.endTime
```

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -44,33 +50,60 @@
+        && postData.value.totalTeachingTimeUnit
+        && postData.value.teachingFee
+        && postData.value.teachingFeeUnit
+        && postData.value.requestTeacherSex 
```
change function isFormValid to
```suggestion
     const isFormValid = computed(() => {
        return isSubjectSelected.value 
        && validateAgeRange()
        && postData.value.shifts.every((shift) => isShiftValid(shift))
        && postData.value.learnerName 
        && postData.value.learnerSex
        && postData.value.learnerbirthYear
        && postData.value.teachingLocation
        && postData.value.totalTeachingTime
        && postData.value.totalTeachingTimeUnit
        && postData.value.teachingFee
        && postData.value.teachingFeeUnit
        && postData.value.requestTeacherSex 
    })
```

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -44,33 +50,60 @@
+            if (response.status < 300) {
+                router.push('/');
+            }
+        }
```
change async function handleSubmit() to
```suggestion
     async function handleSubmit() {
        isFormSubmitted.value = true;
        if (!isFormValid.value) {
            return
        }
        postData.value.subjects = subjectSelected.value;
        const response = await PostService.createPost(postData.value);
        if (response.status < 300) {
            router.push('/');
        }
    }
```

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -288,19 +430,27 @@
+                                currency="VND" 
+                                locale="us-US"
+                                placeholder="Nhập học phí cho từng buổi, tuần hoặc tháng"
+                                :class="['w-100', { 'is-invalid': isFormSubmitted && !postData.teachingFee }]"
```
Anh check lại chỗ này nha, chỗ này sao em không nhập gì mà nó vẫn không báo đỏ

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -223,22 +338,40 @@
+                            id="calendar-endTime" 
+                            v-model="shift.endTime" 
+                            timeOnly 
+                            placeholder="kết thức lúc"
```
Chỗ này kiểm tra lại chính tả nha anh

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -373,55 +555,58 @@
                     <div class="col-lg-4 me-4">
-                        <select class="form-select js-choice z-index-9 border-0 bg-light" aria-label=".form-select-sm" v-model="postData.requestWorkingExp">
+                        <select class="form-select js-choice z-index-9 border-0 bg-light" aria-label=".form-select-sm" v-model="postData.requestTeacherWorkingExp">
                             <option value="" disabled selected>Kinh Nghiệp Làm Việc</option>
```
Check lại chính tả chỗ này nha anh

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -373,55 +555,58 @@
                     <div class="col-lg-4">
-                        <select class="form-select js-choice z-index-9 border-0 bg-light" aria-label=".form-select-sm" v-model="postData.requestTeachingExp">
+                        <select class="form-select js-choice z-index-9 border-0 bg-light" aria-label=".form-select-sm" v-model="postData.requestTeacherTeachingExp">
                             <option value="" disabled selected>Kinh Nghiệp Dạy học</option>
```
Check lại chính tả chỗ này nha anh

rom2kpro commented on src/views/Post/CreatePost.vue:
```diff
@@ -1,40 +1,46 @@
-        teachingMethod: '',
-        subjects: [],
+        teachingLocation: '',
+        subjects: [] as string[],
```
```suggestion
        subjects: [] as any[],
```

rom2kpro commented on src/views/HomeView.vue:
```diff
@@ -5,8 +5,10 @@
     const store = commonStore();
-    onMounted(() => {
-        store.fetchSubjects()
+    onMounted(async () => {
```
If not use this at homepage please remove it

tranVyTDT replied:
home page use it too. filter post/teacher need subjects

rom2kpro commented on src/App.vue:
```diff
@@ -1,6 +1,7 @@
 <!-- eslint-disable prettier/prettier -->
 <script setup lang="ts">
   import { RouterView } from "vue-router";
+  import Toast from 'primevue/toast';
```
Not need import Toast at this

tranVyTDT replied:
fixed

tranVyTDT committed:
Update src/views/Post/CreatePost.vue

const isSubjectSelected = computed(() => {
      return postData.value.selected_subjects.length > 0;
    })

Co-authored-by: viettrung258 <44729642+rom2kpro@users.noreply.github.com>

tranVyTDT committed:
Update src/views/Post/CreatePost.vue

function validateAgeRange() {
        if (!postData.value.requestTeacherAgeFrom || !postData.value.requestTeacherAgeTo ) {
            validationMessage.value = 'Tuổi bên trái và tuổi bên phải phải được chọn cả 2';
                return false
        } else if (postData.value.requestTeacherAgeFrom >= postData.value.requestTeacherAgeTo) {
            validationMessage.value = 'Tuổi bên trái phải nhỏ hơn Tuổi bên phải';
            return false
        } 
        validationMessage.value = '';
        return true
    }

Co-authored-by: viettrung258 <44729642+rom2kpro@users.noreply.github.com>

tranVyTDT committed:
[VyTT][CreatePost] fix cmt

tranVyTDT committed:
Revert "[VyTT][CreatePost] fix cmt"

This reverts commit 6844c9305f5ec55b024a15dd9a8adf30bae53cb9.

tranVyTDT committed:
Revert "Revert "[VyTT][CreatePost] fix cmt""

This reverts commit ff5bb2ed975dbb5736408688607ce4e285b3542f.

tranVyTDT committed:
Update CreatePost.vue

tranVyTDT committed:
Update CreatePost.vue

tranVyTDT committed:
Delete package-lock.json

tranVyTDT committed:
Update CreatePost.vue

tranVyTDT committed:
Merge branch 'add-common-store-for-subject-api' of https://github.com/Vaeteam/front-end into add-common-store-for-subject-api

tranVyTDT committed:
[VyTT][CreatePost] add address

tranVyTDT committed:
[VyTT][CreatePost] update address name

tranVyTDT committed:
Merge branch 'main' into add-common-store-for-subject-api

tranVyTDT committed:
Merge branch 'main' into add-common-store-for-subject-api
